### ISSUE TITLE
Disapearing Contracts

### ISSUE THREAD

**Issue by Yasinahh** [Mmtrx/FS22_BetterContracts#36]
Hello,
I use the mod better contracts and have the error that seed orders always disappear immediately after they have been displayed. You hardly have time to accept them. Orders that have been accepted or completed sometimes disappear immediately after the game starts if the orders have not been completely completed.

**Comment by Yiddan** [comment]
Is your version up to date ?
Did you enable "lazy npc" in the options ?

> "lazy NPCs" (leave more work for contracts) can be configured on/ off

**Comment by Yasinahh** [comment]
Yes, i have updatet it yesterday. 
The error occurred before the update.  I have wrote this ticket before the Update was online in the Modhub.
No, i dont have activate it now

**Comment by rohne83** [comment]
there's a config.xml file in the mod, where you can set the lazy NPCs for different types of field work to true. Per default NPCs are only lazy with harvesting, while for all other types the Giants default behaviour is applied (but even that should leave much more possible contracts for you if you updated your FS version to 1.7.1)

**Comment by Yasinahh** [comment]
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<FS22_BetterContracts debug="false" maxActive="0" lazyNPC="true">
	

	<lazyNPC harvest="true" plowCultivate="false" sow="false" fertilize="false"/> 
	
</FS22_BetterContracts>

So i have to set lazynpc to True? And after that the Filter Harvest, plowCultivate, sow and fertilize is active? For more fertilize, sow nd plowcultivating contracts i have to set it true too?

**Comment by Mmtrx** [comment]
That is completely correct 👍

**Comment by Mmtrx** [comment]
Oh, and you should make those edits in file /modSettings/FS22_BetterContracts.xml 
Not in Config.xml inside the mods zip file.

**Comment by Yasinahh** [comment]
perfect! Thanks a lot! :-)

**Comment by FatLizard** [comment]
@Mmtrx Is there a way to offer you a coffee ? ... ;)

**Comment by PapaTim68** [comment]
Just updated to the newest Version. I have the same Issue and I made sure to change the config.
The Contracts disappear instantly when refreshing contracts. Its happening to sowing and ferterlizing contracts based on the Vehicles that I can see for a split second, they are visible for a mili second or so and then gone and finished it seems like because the field status changes.
One thing of note I am on a Dedicated Server and also have Supply Contracts mod installed.
![grafik](https://user-images.githubusercontent.com/29522870/192600914-f18ae749-9a18-40d0-9c35-1fabd6981aa9.png)

**Comment by Yasinahh** [comment]
The orders still disappear after a short time. I'm now on a different map and the error persists.

**Comment by Mmtrx** [comment]
Could you also verify that with `lazyNPC="false"`? If yes, it's a basegame problem. 
What is your seasons setting, 1 day / month? (I have not tested yet with seasons off.)

Also, there is no need to first clear all contracts, if you want some new ones: just keep hitting "X" until no more contracts are generated. Then you have the max available number of contracts for that month.

**Comment by PapaTim68** [comment]
It happens with both true and false for lazyNPC. 6 days/month with seasonal growth. Happens with and without prior clearing. With seasons on. One thing noticed we now changed months from February to March and Contracts started saying, but also some contracts definitely were missed and done by the NPC in Feb. I looked at it with debug on. When updating it gives the refresh() debug 5-6 Contracts 4 of which are Supply Contracts and instantly after that refresh() debug with the 4 Supply Contracts.
